cmd/gox/template/project/.builds/web/game.js

```diff
@@ -552,4 +549,167 @@ class GameApp {
+
+        const processedArgs = [];
+        let callbackCounter = 0;
+
+        for (let arg of args) {
+            if (typeof arg === 'function') {
+                // generate unique callback name
+                const callbackName = `_onSpxCall_${Date.now()}_${callbackCounter++}`;
+                
+                // register callback function
+                this.registerWorkerCallback(callbackName, arg);
+                
+                // replace with main thread callback identifier
+                processedArgs.push("_SPX_CALLBACK_FUNC_", callbackName);
+            } else {
+                processedArgs.push(arg);
+            }
+        }
+
+        return processedArgs;
+    }
+
+    // register worker callback function
+    registerWorkerCallback(callbackName, userFunction) {
+        // create callback handler function
+        window._spxMainCalls[callbackName] = async function(requestId, ...args) {
+            let errorMsg = null;
+            let result = null;
+            
+            try {
+                if (userFunction) {
+                    result = userFunction(...args);
+                    // if return Promise, wait for it to complete
+                    if (result && typeof result.then === 'function') {
+                        result = await result;
+                    }
+                } else {
+                    errorMsg = `No function registered for ${callbackName}`;
+                }
+            } catch (error) {
+                console.error(`Error in ${callbackName}:`, error);
+                errorMsg = error.message;
+            }
+            
+            // send response to worker
+            this.postMessageToWorkers({
+                cmd: 'callResponse',
+                responseId: requestId,
+                result: result,
+                error: errorMsg
+            });
+        }.bind(this);
+    }
+
+    postMessageToWorkers(message, transferList = null, cloneForEach = false) {
+        const workers = [];
+        if (this.pthreads) {
+            workers.push(...this.pthreads.runningWorkers);
+        }
+
+        let successCount = 0;
+        let errorCount = 0;
+        
+        workers.forEach((worker, index) => {
+            try {
+                if (worker && typeof worker.postMessage === 'function') {
+                    // Adds unique identifier and target info to each message
+                    let enhancedMessage = {
+                        ...message,
+                        _gameAppMessageId: ++this.workerMessageId,
+                        _targetWorkerIndex: index,
+                        _timestamp: Date.now()
+                    };
+                    
+                    // Special handling required when cloning data or using transferList
+                    if (transferList && cloneForEach) {
+                        if (message.data && message.data.buffer) {
+                            const clonedData = new Uint8Array(message.data);
```

Comment:
AFAIK we don't need to copy the buffer data by constructing a new `Uint8Array` instance. It will be copied automatically when transferred by `postMessage`. See details in https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer#transferring_arraybuffers